sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -3547,6 +3616,110 @@ def value_counts(self, subset=None, sort=False, normalize=False,
         return result
 
 
+  @frame_base.with_docs_from(pd.DataFrame)
+  @frame_base.args_to_kwargs(pd.DataFrame)
+  @frame_base.populate_defaults(pd.DataFrame)
+  def idxmin(self, **kwargs):
+    axis = kwargs.get('axis', 0)
+
+    index_dtype = self._expr.proxy().index.dtype
+    columns_dtype = self._expr.proxy().columns.dtype
+
+    def compute_idxmin(df):
+      min_indexes = df.idxmin(**kwargs).unique()
+      if pd.isna(min_indexes).any():
+        return df
+      else:
+        return df.loc[min_indexes]
+
+    if axis in ('index', 0):
+      requires_partition = partitionings.Singleton()
+
+      proxy_index = pd.Index([], dtype=columns_dtype)
+      proxy = pd.Series([], index=proxy_index, dtype=index_dtype)
+      partition_proxy = self._expr.proxy().copy()
+
+      idxmin_per_partition = expressions.ComputedExpression(
+        'idxmin-per-partition',
+        compute_idxmin, [self._expr],
+        proxy=partition_proxy,
+        requires_partition_by=partitionings.Arbitrary(),
+        preserves_partition_by=partitionings.Arbitrary()
+      )
+
+    elif axis in ('columns', 1):
+      requires_partition=partitionings.Index()
+
+      proxy_index = pd.Index([], dtype=index_dtype)
+      proxy = pd.Series([], index=proxy_index, dtype=columns_dtype)
+
+      idxmin_per_partition = self._expr
+
+
+    with expressions.allow_non_parallel_operations(True):
+      return frame_base.DeferredFrame.wrap(
+        expressions.ComputedExpression(
+          'idxmin',
+          lambda df: df.idxmin(**kwargs), [idxmin_per_partition],
+          proxy=proxy,
+          requires_partition_by=requires_partition,
+          preserves_partition_by=partitionings.Singleton()
+        )
+      )
+
+
+  @frame_base.with_docs_from(pd.DataFrame)
+  @frame_base.args_to_kwargs(pd.DataFrame)
+  @frame_base.populate_defaults(pd.DataFrame)
+  def idxmax(self, **kwargs):
```

Comment:
You might consider adding a helper (here and in Series) like `def _idxmaxmin_helper(self, op, kwargs):` (where op is 'idxmin' or 'idxmax'). That could let you implement the common logic in one place. I'll leave that up to you though.